libraries/ms-drivers/inc/adt_7470_fan_controller.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "adc.h"
+#include "gpio.h"
+#include "i2c.h"
+#include "soft_timer.h"
+
+typedef void (*Adt7470DataCallback)();
+
+typedef struct {
+  I2CPort i2c_port;
+  GpioAddress *fan_1_pin;
+  GpioAddress *fan_2_pin;
+  GpioAddress *fan_3_pin;
+  GpioAddress *fan_4_pin;
+  uint32_t interval_us;
+  Adt7470DataCallback callback;  // set to NULL for no callback
+  void *callback_context;
+} Adt7470Settings;
+
+typedef struct {
+  GpioAddress *fan_1_pin;
+  GpioAddress *fan_2_pin;
+  GpioAddress *fan_3_pin;
+  GpioAddress *fan_4_pin;
+  uint32_t interval_us;
+  SoftTimerId timer_id;
+  Adt7470DataCallback callback;
+  void *callback_context;
+} Adt7470Storage;
+
+// probably a better name for these
+typedef enum { FAN_1 = 0, FAN_2, FAN_3, FAN_4 } Fan;
+
+// Initialize the Adt7200 with the given settings; the select pin is an STM32 GPIO pin.
+StatusCode adt_7200_init_stm32(Adt7470Storage *storage, Adt7470Settings *settings);
```

Comment:
should be fine to just call it `adt7470_init`, no need for stm32. Also, where did 7200 come from?